smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/serde/xml/MemberShapeDecodeXMLGenerator.kt

```diff
@@ -43,51 +43,60 @@ abstract class MemberShapeDecodeXMLGenerator(
     fun renderListMember(
         member: MemberShape,
         memberTarget: CollectionShape,
-        containerName: String
+        containerName: String,
+        isUnion: Boolean = false
```

Comment:
1.  Rather than pass in isUnion to each of the functions, it seems more appropriate to add the following to MemberShapeDecodeXMLGenerator:
```
    abstract val isUnion: Boolean
```
2. The public functions that were modified to accept `isUnion` (e.g: `renderListMember`, `renderMapMember`, etc.) seem to be completely different between a union and a struct.  For that reason, it seems to make sense to put unique functions into UnionDecodeXMLGenerator and StructDecodeXMLGenerator rather than having a big if/else in the abstract class.  Doing so, would allow us to avoid creating an `abstract val isUnion`.